# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh
# Commit: fb720435215f
# Full Hash: fb720435215fa5be3b2bd2101f3a54382954b70c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1544232
# File Overlap Count: 4
# Description:
#   Bug 1544232 - Limit lifetime extension of SWs by SWs to the sender's lifetime. r=edenchuang
#   
#   This patch introduces an explicit concept of lifetimes with mechanisms
#   in place so that actions taken by Clients (windows or non-ServiceWorker
#   orkers) will extend the lifetime of a ServiceWorker, but a ServiceWorker
# ==============================================================================

diff -r 25452f6623a0 -r fb720435215f dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh
--- a/dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh	Thu Oct 24 03:02:41 2024 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh	Thu Oct 24 03:02:42 2024 +0000
@@ -67,6 +67,14 @@
 
   bool isSecureContext;
 
+  // For SharedWorkers this serves as the loadingClientInfo for the script loads
+  // and the basis for CSP for the global.
+  //
+  // For ServiceWorkers this is pre-created ClientSource information which
+  // should be used instead of a freshly created ClientSource.  This is done so
+  // that a ServiceWorker's client id can have a mapping established in the
+  // parent so that the keepalive contribution of ServiceWorkers to other
+  // ServiceWorkers is limited to their own keepalive lifetime.
   IPCClientInfo? clientInfo;
 
   nullable nsIReferrerInfo referrerInfo;